### PR TITLE
feat(RND-13624): Migrate `WithS3File` to AWS Data Wrangler

### PR DESCRIPTION
**Pull request by Christos-Hadjinikolis**
## Links
RND-13624

#### Summary
This PR refactors the `WithS3File` mixin to use [`[awswrangler](https://github.com/aws/aws-sdk-pandas)`](https://github.com/aws/aws-sdk-pandas) for all S3 I/O operations (CSV, JSON, Parquet, and HDF), removing reliance on local temp files, `boto3`, and the AWS CLI. HDF files are now streamed in-memory using `boto3.client().download_fileobj` and PyTables.

#### Key Changes
- ✅ Migrated all S3 reads/writes to use `awswrangler`:
  - CSV, JSON, and Parquet are handled via direct `wr.s3.read_*/to_*` calls.
  - HDF files are streamed to memory using `BytesIO`.
- ✅ Refactored I/O logic into `_read_s3_{type}_file` / `_write_s3_{type}_file` methods.
- ✅ Updated `allow_options` decorator to:
  - Filter passthrough kwargs per backend (`pandas`, `pyarrow`, `wrangler`, etc).
  - Support argument containers like `pandas_kwargs`, `pyarrow_additional_kwargs`, `s3_additional_kwargs`.
- ✅ Blocked automatic datetime parsing in JSON readers:
  - `convert_dates=True` is ignored to ensure consistency.
- ✅ Introduced support for single-record JSON objects:
  - Added `single_record=True` option to force wrapping a single JSON object into a tabular DataFrame.
  - Detected and warned on suspiciously structured inputs (e.g. when pandas misparses a dict as a 1-column/2-row DataFrame).
- ✅ Ensured full consistency between `pandas.read_json` and `awswrangler.read_json` for `orient="records"` and `lines=True` usage.
- 🧪 Extended and updated test coverage, including:
  - Ensuring only schema-specified columns are loaded.
  - Verifying column order is preserved.
  - Warning is raised for likely single-record JSONs when option is missing.
  - Confirming timestamps are **not** auto-parsed.

#### Benefits
- 🚀 Improved performance and native AWS integration via `awswrangler`.
- 🧼 Cleaner, more maintainable, and safer I/O logic.
- 🧪 Easier to test (no tmp files, no shell commands).
- 🔍 Transparent, configurable behavior for edge cases like JSON formatting.

#### Notes
- Legacy S3 logic (`_s3_reader`, `_s3_writer`, `_s3_named_file_reader`) has been fully removed.
- `awswrangler` is now a required dependency.
- Developers can enable consistent behavior for single-record JSON inputs via `single_record=True`.

### CODE REVIEW

File: dynamicio/__main__.py [hunk 1]
```diff
@@ -1,4 +1,6 @@
 """Invokes dynamicio cli."""
+
+# Application Imports
```
Comment (Christos-Hadjinikolis) [inline dynamicio/__main__.py:3]: Black does this now...

File: dynamicio/mixins/with_athena.py [hunk 2]
```diff
@@ -19,7 +19,15 @@
...
-    """Handles I/O operations for AWS Athena."""
+    """Handles I/O operations for AWS Athena.
+
+    Note:
+        The `__abstractmethods__ = frozenset()` is used to silence false positives from pylint,
+        which might wrongly assume this class has abstract methods due to NotImplementedError.
+        This class is *not* an abstract base class and does not use `abc.ABC`.
+    """
+
+    __abstractmethods__ = frozenset()
```
Comment (Christos-Hadjinikolis) [inline dynamicio/mixins/with_athena.py:30]: Fixes a `pylint` false positive on abstract methods not implemented...

File: dynamicio/mixins/with_local.py [hunk 3]
```diff
@@ -106,23 +110,49 @@ def _read_csv_file(file_path: str, schema: DataframeSchema, **options: Any) -> p
...
+            # Re-wrap as single dict row — i.e., rehydrate the record
+            df = pd.DataFrame([{df.columns[0]: dict(zip(df.index, df.iloc[:, 0]))}])
+        elif (
+            df.shape[1] == 1
+            and df.columns.dtype == "object"
+            and df.index.dtype == "object"
+            and all(isinstance(i, str) for i in df.index)
+            and all(isinstance(v, (str, int, float, bool, type(None))) for v in df.iloc[:, 0])
+        ):
+            logger.warning("[local-json-read] File appears to be a single-record JSON object. Pass 'single_record=True' in options to handle this case.")
```
Comment (Christos-Hadjinikolis) [inline dynamicio/mixins/with_local.py:153]: Necessary for alignment with `wrangler.read_json()` behaviour.

**Review by yuan-vortexa — DISMISSED** [review]
lgtm
